### PR TITLE
Save fast-reboot state into the db

### PR DESCRIPTION
**Pull request by pavel-shirshov**
Fixes https://github.com/Azure/sonic-buildimage/issues/3697

**- What I did**
1. Put a flag for fast-reboot to the db
2. Using this flag in other part of SONiC to start in Fast-reboot mode

**- How I did it**
1. Using EXPIRE feature of db
2. If we reload a config, the state in the db will be removed.

**- How to verify it**
Build your DUT and test on your dut.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by yxieca** [comment]
Looks generally good to me. I am wondering what is your plan of using the EX 180 information?

We might want to run a test fast-reboot -> make sure it is executed as fast-reboot. Then restart swss, make sure it is a cold restart.

**Comment by pavel-shirshov** [comment]
@yxieca This is instruction to redis to remove the key in 180 seconds.

**Comment by yxieca** [comment]
I see. Interesting. Do we need this expiry? swss restart should clear the db including this key, right?

**Comment by pavel-shirshov** [comment]
@yxieca But if we restart bgpd only it will be always fast-reboot start of bgpd

**Comment by yxieca** [comment]
Good point!

**Comment by pavel-shirshov** [comment]
retest broadcom please

**Comment by pavel-shirshov** [comment]
retest vsimage please

**Comment by yxieca** [comment]
retest vsimage please

**Comment by yxieca** [comment]
retest broadcom please

**Comment by pavel-shirshov** [comment]
retest vs please

**Comment by pavel-shirshov** [comment]
retest broadcom please

**Comment by pavel-shirshov** [comment]
retest vs please

File: platform/broadcom/docker-syncd-brcm/Dockerfile.j2 [hunk 1]
```diff
@@ -14,6 +14,9 @@ debs/{{ deb }}{{' '}}
 {%- endfor -%}
 debs/
 
+# redis-tools dependencies
+RUN apt-get install -y libatomic1 libjemalloc1 liblua5.1-0 lua-bitop lua-cjson
```
Comment (lguohan) [inline platform/broadcom/docker-syncd-brcm/Dockerfile.j2]: can you put this in the docker-syncd-base.mk?

File: platform/innovium/docker-syncd-invm.mk [hunk 2]
```diff
@@ -3,7 +3,7 @@
 DOCKER_SYNCD_PLATFORM_CODE = invm
 include $(PLATFORM_PATH)/../template/docker-syncd-base.mk
 
-$(DOCKER_SYNCD_BASE)_DEPENDS += $(SYNCD) $(PYTHON_SDK_API) $(INVM_LIBSAI)
+$(DOCKER_SYNCD_BASE)_DEPENDS += $(SYNCD) $(PYTHON_SDK_API) $(INVM_LIBSAI) $(REDIS_TOOLS)
```
Comment (lguohan) [inline platform/innovium/docker-syncd-invm.mk]: this is not correct. should not be DOCKER_SYNCD_BASE.

**Comment by pavel-shirshov** [comment]
retest vsimage please

**Comment by pavel-shirshov** [comment]
retest vsimage please

**Comment by pavel-shirshov** [comment]
retest vmimage please

**Comment by pavel-shirshov** [comment]
retest vsimage please

**Comment by lguohan** [comment]
can you rebase and resolve conflict?

**Comment by lguohan** [comment]
for master we are using frr, so sonic-frr init should be modified. I will let pavel fix that.

**Comment by yxieca** [comment]
Separate PR was created for 201811 branch due to merge conflicts.

**Pull request merged**